msal/src/test/java/com/microsoft/identity/client/nativeauth/NativeAuthPublicClientApplicationKotlinTest.kt

```diff
@@ -0,0 +1,2011 @@
+            MockApiResponseType.INITIATE_SUCCESS
+        )
+
+        configureMockApi(
+            MockApiEndpoint.SignInChallenge,
+            correlationId,
+            MockApiResponseType.CHALLENGE_TYPE_PASSWORD
+        )
+
+        configureMockApi(
+            endpointType = MockApiEndpoint.SignInToken,
+            correlationId = correlationId,
+            responseType = MockApiResponseType.TOKEN_SUCCESS
+        )
+
+        val result = application.signInUsingPassword(username, password)
+        assertTrue(result is SignInResult.Complete)
+
+        try {
+            application.signInUsingPassword(username, password)
+        } catch (exception: MsalException) {
+            assertEquals(MsalClientException.INVALID_PARAMETER, exception.errorCode)
+            assertEquals("An account is already signed in.", exception.message)
+            return@runTest
+        }
+        fail() // An exception should happen
+    }
+
+    /**
+     * Test sign in with SLT scenario 1:
+     * 1a -> sign in with (valid) SLT
+     * 1b <- server returns token
+     */
+    @Test
+    fun testSignInWithSLT() = runTest {
+        // Setup - sign up the user, so that we don't have to construct the SLT state manually
+        // as this doesn't allow for the NativeAuthPublicClientApplicationConfiguration to be set
+        // up, meaning it would need to be mocked (which we don't want in these tests).
+        val signInWithSLTState = signUpUser()
+
+        // 1a. sign in with (valid) SLT
+        val correlationId = UUID.randomUUID().toString()
+        configureMockApi(
+            endpointType = MockApiEndpoint.SignInToken,
+            correlationId = correlationId,
+            responseType = MockApiResponseType.TOKEN_SUCCESS
+        )
+
+        // 1b. server returns token
+        val result = signInWithSLTState.signIn(scopes = null)
+        assertTrue(result is SignInResult.Complete)
+    }
+
+    /**
+     * Test sign in with SLT scenario 2:
+     * 1a -> sign in with (null) SLT
+     * 1b <- client returns error right away
+     */
+    @Test
+    fun testSignInWithSLTNullSLT() = runTest {
+        // 1a. sign in with (null) SLT
+        val correlationId = UUID.randomUUID().toString()
+        configureMockApi(
+            endpointType = MockApiEndpoint.SignInToken,
+            correlationId = correlationId,
+            responseType = MockApiResponseType.TOKEN_SUCCESS
+        )
+
+        // 1b. client returns error
+        val config = mock<NativeAuthPublicClientApplicationConfiguration>()
+        val sltState = SignInAfterSignUpState(signInVerificationCode = null, username = username, config = config)
+        val result = sltState.signIn(scopes = null)
+        assertTrue(result is SignInResult.UnexpectedError)
+    }
+
+    /**
+     * Test sign in with SLT scenario 2:
+     * 1a -> sign in with (expired) SLT
+     * 1b <- server returns error
+     */
+    @Ignore("Waiting for SLT Mock API Integration (Out of scope?)")
```

Comment:
Same here, do we have mock api support for slt yet

Reply:
The SLT support is in the mock APIs. I will enable this test.